lib/router/route-info.ts

```diff
@@ -63,9 +65,10 @@ export function toReadOnlyRouteInfo(
   includeAttributes = false
 ): RouteInfoWithAttributes[] | RouteInfo[] {
   return routeInfos.map((info, i) => {
-    let { name, params, paramNames, context } = info;
+    let { name, params, paramNames, context, route } = info;
     if (ROUTE_INFOS.has(info) && includeAttributes) {
       let routeInfo = ROUTE_INFOS.get(info)!;
+      routeInfo = attachMetadata(route!, routeInfo);
```

Comment:
Why do we need to assert `route` is non-null here? Is it guaranteed to be populated by this phase of the lifecycle?

Reply:
Yes, it is in a resolved state. This function is also used for the pre-resolved state.